java/short_url_generator/Index.java

```diff
@@ -0,0 +1,280 @@
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.*;
+import java.util.stream.*;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import com.google.gson.Gson;
+import org.apache.commons.validator.routines.UrlValidator;
+
+// List of supported providers
+private enum Provider {
+
+    BITLY("bitly"),
+    TINY_URL("tinyurl");
+
+    private final String name;
+
+    Provider(String name) {
+        this.name = name;
+    }
+
+    static boolean validateProvider(String name) {
+        for(Provider providerName: Provider.values()) {
+            if(providerName.name.equals(name)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    String getName() {
+        return name;
+    }
+}
+
+final Gson gson = new Gson();
+
+final Map<String, String> endpointsMap = Map.of(
+    "bitly", "https://api-ssl.bitly.com/v4/shorten",
+    "tinyurl", "https://api.tinyurl.com/create"
+);
+
+public RuntimeResponse main(RuntimeRequest req, RuntimeResponse res) throws Exception {
+    
+    // Validate that values present in the request are not empty (payload, variables)
+    RuntimeResponse errorResponse = checkEmptyPayloadAndVariables(req, res);
+    if(errorResponse != null) {
+        return errorResponse;
+    }
+
+    // Validate that the API key is not empty
+    errorResponse = checkEmptyAPIKey(req, res);
+    if(errorResponse != null) {
+        return errorResponse;
+    }
+
+    String apiKey = req.getVariables().get("apiKey");
```

Comment:
This will need to be changed to either get `TINYURL_API_KEY` or `BITLY_API_KEY`.